### ISSUE TITLE
[java] AccessorClassGeneration false positive for private constructors

### ISSUE THREAD

**Issue by dotdoom** [pmd/pmd#755]
**Rule Set:**
Design: AccessorClassGeneration

**Description:**
When there are both private and public constructors in inner class, with the same number of arguments but different types, PMD assumes that private is called from the outer class even public is actually called.

**Code Sample demonstrating the issue:**

```java
public class Outer
{
  public static class Inner {
    private Inner(int x) { }
    public Inner(Object x) { }
  }
  
  public static void main(String[] args)
  {
    // PMD.AccessorClassGeneration
    // Avoid instantiation through private constructors from outside of the constructors class.
    new Inner(null); // complains as if Inner(int) is was called
  }
}
```

**Running PMD through:** *[Gradle]*

**Comment by dotdoom** [comment]
Aww, I think there's already a TODO for this:

https://pmd.github.io/pmd-5.8.1/pmd-java/xref/net/sourceforge/pmd/lang/java/rule/design/AccessorClassGenerationRule.html#L87

**Comment by jsotuyod** [comment]
@dotdoom thanks for the feedback! You are absolutely right on this one. That TODO is not exactly related to your issue, but refers to a time where type resolution support was non-existing.

You are absolutely right about this one. 6.0.0 is already too close to release, but this should definitely make it into 6.0.1

**Comment by oowekyala** [comment]
Was fixed in #2899
